### PR TITLE
dotCMS/core#18678 Unable to show the push publish modal from edit mode

### PR DESCRIPTION
**Pull request by hmoreras**
Wizard feedback.

### CODE REVIEW

File: src/app/api/services/dot-roles/dot-roles.service.ts [hunk 1]
```diff
@@ -21,16 +21,16 @@ export class DotRolesService {
      * @returns Observable<DotRole[]>
      * @memberof DotRolesService
      */
-    get(roleId: string): Observable<DotRole[]> {
+    get(roleId: string, roleHierarchy: boolean): Observable<DotRole[]> {
```
Comment (fmontes) [inline src/app/api/services/dot-roles/dot-roles.service.ts:24]: Why don't we leave a default here?
```suggestion
    get(roleId: string, roleHierarchy = true): Observable<DotRole[]> {
```

File: src/app/api/services/dot-roles/dot-roles.service.ts [hunk 2]
```diff
@@ -21,16 +21,16 @@ export class DotRolesService {
...
             .requestView({
                 method: RequestMethod.Get,
-                url: `/api/v1/roles/${roleId}/rolehierarchyanduserroles`
+                url: `/api/v1/roles/${roleId}/rolehierarchyanduserroles?roleHierarchyForAssign=${roleHierarchy}`
             })
             .pipe(
                 pluck('entity'),
                 map((roles: DotRole[]) =>
-                    roles.filter(role => role.user === false).map((role: DotRole) => {
+                    roles.filter(role => role.roleKey !== 'anonymous').map((role: DotRole) => {
```
Comment (fmontes) [inline src/app/api/services/dot-roles/dot-roles.service.ts:33]: type?

File: src/app/api/services/dot-workflows-actions/dot-workflows-actions.service.spec.ts [hunk 3]
```diff
@@ -130,7 +131,7 @@ describe('DotWorkflowsActionsService', () => {
         const mockWorkflowActions: DotCMSWorkflowAction = {
             ...mockWorkflowsActions[0]
         };
-        let steps: DotWizardStep<any>[];
+        let wizardInpnut: DotWizardInput;
```
Comment (fmontes) [inline src/app/api/services/dot-workflows-actions/dot-workflows-actions.service.spec.ts:134]:
```suggestion
        let wizardInput: DotWizardInput;
```

File: src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.spec.ts [hunk 4]
```diff
@@ -30,6 +37,12 @@ import { PushPublishService } from '@services/push-publish/push-publish.service'
 import { MockPushPublishService } from '@portlets/shared/dot-content-types-listing/dot-content-types.component.spec';
 import { DotMessageDisplayService } from '@components/dot-message-display/services';
 import { DotMessageSeverity, DotMessageType } from '@components/dot-message-display/model';
+import { BaseRequestOptions, ConnectionBackend, Http, RequestOptions } from '@angular/http';
+import { MockBackend } from '@angular/http/testing';
+import { CoreWebServiceMock } from '../../../../../../../projects/dotcms-js/src/lib/core/core-web.service.mock';
```
Comment (fmontes) [inline src/app/portlets/dot-edit-page/content/components/dot-edit-page-workflows-actions/dot-edit-page-workflows-actions.component.spec.ts:42]: Is this ok?

File: src/app/api/services/dot-workflows-actions/dot-workflows-actions.service.spec.ts [hunk 5]
```diff
@@ -18,6 +18,7 @@ import { DotCommentAndAssignFormComponent } from '@components/_common/forms/dot-
 import { DotPushPublishFormComponent } from '@components/_common/forms/dot-push-publish-form/dot-push-publish-form.component';
 import { CoreWebService } from 'dotcms-js';
 import { CoreWebServiceMock } from '../../../../../projects/dotcms-js/src/lib/core/core-web.service.mock';
```
Comment (fmontes) [inline src/app/api/services/dot-workflows-actions/dot-workflows-actions.service.spec.ts:20]: is this ok?

File: src/app/api/services/push-publish/push-publish.service.spec.ts [hunk 6]
```diff
@@ -1,11 +1,19 @@
...
+    BaseRequestOptions,
+    Http
+} from '@angular/http';
 import { MockBackend } from '@angular/http/testing';
-import { fakeAsync, tick } from '@angular/core/testing';
+import { fakeAsync, TestBed, tick } from '@angular/core/testing';
 import { DotCurrentUserService } from '../dot-current-user/dot-current-user.service';
 import { DotPushPublishData } from '@models/dot-push-publish-data/dot-push-publish-data';
+import { ApiRoot, CoreWebService, LoggerService, StringUtils, UserModel } from 'dotcms-js';
+import { CoreWebServiceMock } from '../../../../../projects/dotcms-js/src/lib/core/core-web.service.mock';
```
Comment (fmontes) [inline src/app/api/services/push-publish/push-publish.service.spec.ts:16]: is this ok?

File: src/app/shared/models/ajax-action-response/ajax-action-response.ts [hunk 7]
```diff
@@ -4,6 +4,7 @@
  * @interface
  */
 export interface AjaxActionResponseView {
```
Comment (fmontes) [inline src/app/shared/models/ajax-action-response/ajax-action-response.ts:6]:
```suggestion
export interface DotAjaxActionResponseView {
```

File: src/app/api/services/dot-wizard/dot-wizard.service.ts [hunk 8]
```diff
@@ -26,8 +26,8 @@ export class DotWizardService {
      * @returns Observable<{ [key: string]: string }>
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-wizard/dot-wizard.service.ts:26]: fix doc

File: src/app/api/services/dot-workflows-actions/dot-workflows-actions.service.ts [hunk 9]
```diff
@@ -59,13 +60,13 @@ export class DotWorkflowsActionsService {
     }
 
     /**
-     * Returns the steps needed to collect information of the Workflow inputs.
+     * Returns the input needed to collect information of the Workflow inputs.
      *
      * @param {DotCMSWorkflowAction} workflow
      * @returns DotWizardStep[]
```
Comment (alfredo-dotcms) [inline src/app/api/services/dot-workflows-actions/dot-workflows-actions.service.ts:66]: fix doc

File: src/app/api/services/dot-roles/dot-roles.service.ts (hunk 1 above)
Comment (hmoreras) [reply src/app/api/services/dot-roles/dot-roles.service.ts:24]: this is taken from the workflow given by the endpoint.

**Pull request merged**